### PR TITLE
Fix attribute color list  doesn't show on product list if turning on smarty cache 

### PR DESCRIPTION
**Pull request by TiaNex-Com**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.x
| Description?  | attribute color list  doen't show on product list if turning on smarty cache 
| Type?         | bug fix 
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-6778
| How to test?  | turn on smarty cache

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1654,24 +1654,28 @@ public function addColorsToProductList(&$products)
...
-                    'link'        => Context::getContext()->link,
-                    'img_col_dir' => _THEME_COL_DIR_,
-                    'col_img_dir' => _PS_COL_IMG_DIR_
-                ));
-            }
-
-            if (!in_array($product['id_product'], $products_need_cache) || isset($colors[$product['id_product']])) {
-                $product['color_list'] = $tpl->fetch(_PS_THEME_DIR_.'product-list-colors.tpl', $this->getColorsListCacheId($product['id_product']));
+            if (in_array($product['id_product'], $products_need_cache)) { 
+                $tpl = $this->context->smarty->createTemplate(_PS_THEME_DIR_.'product-list-colors.tpl', $this->getColorsListCacheId($product['id_product']));
```
Comment (jocel1) [inline classes/controller/FrontController.php]: move this line inside the if() condition below

File: classes/controller/FrontController.php [hunk 2]
```diff
@@ -1654,24 +1654,28 @@ public function addColorsToProductList(&$products)
...
+                if (isset($colors[$product['id_product']])) {
+                    $tpl->assign(array(
+                        'id_product'  => $product['id_product'],
+                        'colors_list' => $colors[$product['id_product']],
+                        'link'        => Context::getContext()->link,
+                        'img_col_dir' => _THEME_COL_DIR_,
+                        'col_img_dir' => _PS_COL_IMG_DIR_
+                    ));
+                }
+                if (isset($colors[$product['id_product']])) {
```
Comment (jocel1) [inline classes/controller/FrontController.php]: this if() is the same than the previous one, remove it and move the $product['color_list'] assignation

File: classes/controller/FrontController.php [hunk 3]
```diff
@@ -1654,24 +1654,28 @@ public function addColorsToProductList(&$products)
...
+                    $product['color_list'] = '';
+                }
             } else {
-                $product['color_list'] = '';
+                $tpl = $this->context->smarty->createTemplate(_PS_THEME_DIR_.'product-list-colors.tpl', $this->getColorsListCacheId($product['id_product']));
+                $product['color_list'] = $tpl->fetch(_PS_THEME_DIR_.'product-list-colors.tpl', $this->getColorsListCacheId($product['id_product']));
             }
         }
-        Tools::restoreCacheSettings();
+        Tools::restoreCacheSettings();
```
Comment (jocel1) [inline classes/controller/FrontController.php]: remove extra spaces

**Review by jocel1 — CHANGES_REQUESTED** [review]

File: classes/controller/FrontController.php (hunk 2 above)
Comment (iTiaNex) [inline classes/controller/FrontController.php]: yes,thank you

**Review by TiaNex-Com — COMMENTED** [review]
OK,thank you

**Comment by iTiaNex** [comment]
sorry,it seems it still didn't fix the bug,

**Comment by Casper-O** [comment]
thanks @iTiaNex I first tested your PR but that quickly showed not to work (as you said), now i'm testing your code from your comment and so far it has been keeping the colors for the last 5 mins (and still counting). I am testing it on a 1.6.1.16 where colors disappear from the product-list in categories and manufacturer list of products.

**Pull request closed without merging**